elasticgraph-graphql/lib/elastic_graph/graphql/filtering/filter_interpreter.rb

```diff
@@ -36,11 +36,25 @@ def initialize(filter_node_interpreter:, logger:)
         # Returns `nil` if there are no query clauses, to make it easy for a caller to `compact` out
         # `query: {}` in a larger search request body.
         #
+        # When multiple filter hashes are provided (e.g. client_filters + internal_filters),
+        # process each into its own bool query so they are ANDed together. This ensures
+        # that range queries from different sources intersect instead of override.
+        #
         # https://www.elastic.co/guide/en/elasticsearch/reference/8.11/query-dsl.html
         def build_query(filter_hashes, from_field_path: FieldPath.empty)
-          build_bool_hash do |bool_node|
-            filter_hashes.each do |filter_hash|
-              process_filter_hash(bool_node, filter_hash, from_field_path)
+          if filter_hashes.size <= 1
+            build_bool_hash do |bool_node|
+              filter_hash = filter_hashes.first
+              process_filter_hash(bool_node, filter_hash, from_field_path) if filter_hash
+            end
+          else
+            build_bool_hash do |outer_bool_node|
+              filter_hashes.each do |filter_hash|
+                sub_query = build_bool_hash do |inner_bool_node|
+                  process_filter_hash(inner_bool_node, filter_hash, from_field_path)
+                end
+                outer_bool_node[:filter] << sub_query if sub_query
+              end
             end
```

Comment:
```suggestion
          build_bool_hash do |outer_bool_node|
            filter_hashes.each do |filter_hash|
              sub_query = build_bool_hash do |inner_bool_node|
                process_filter_hash(inner_bool_node, filter_hash, from_field_path)
              end

              outer_bool_node[:filter] << sub_query if sub_query
            end

            if (resulting_filters = outer_bool_node[:filter]).size < 2
              return resulting_filters.first
            end
          end
```

I like this better--instead of treating `<= 1` as a completely different case from 2 or more filters it does the exact same thing (builds up an outer bool node with subfilters). But then just before returning, it checks to see if it can just return a single inner node if it's safe to do so.

I find it easier to reason about when the scope of conditionals is narrowed so that the different paths taken are minimized.